src/Microsoft.Health.Fhir.Api/Features/BackgroundTaskService/TaskHostingBackgroundService.cs

```diff
@@ -0,0 +1,73 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Extensions.DependencyInjection;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.TaskManagement;
+
+namespace Microsoft.Health.Fhir.Api.Features.BackgroundTaskService
+{
+    /// <summary>
+    /// The background service used to host the <see cref="TaskHosting"/>.
+    /// </summary>
+    public class TaskHostingBackgroundService : BackgroundService
+    {
+        private readonly Func<IScoped<TaskHosting>> _taskHostingFactory;
+        private readonly TaskHostingConfiguration _taskHostingConfiguration;
+        private readonly ILogger<TaskHostingBackgroundService> _logger;
+
+        public TaskHostingBackgroundService(
+            Func<IScoped<TaskHosting>> taskHostingFactory,
+            IOptions<TaskHostingConfiguration> taskHostingConfiguration,
+            ILogger<TaskHostingBackgroundService> logger)
+        {
+            EnsureArg.IsNotNull(taskHostingFactory, nameof(taskHostingFactory));
+            EnsureArg.IsNotNull(taskHostingConfiguration?.Value, nameof(taskHostingConfiguration));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _taskHostingFactory = taskHostingFactory;
+            _taskHostingConfiguration = taskHostingConfiguration.Value;
```

Comment:
same~ Is it safer to check if it is null? taskHostingConfiguration?.Value

Reply:
Same here: EnsureArg.IsNotNull(taskHostingConfiguration?.Value, nameof(taskHostingConfiguration));